Tools/scripts/run_astyle.py

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env python
+
+"""
+Runs astyle over directory sub-trees known to be "astyle-clean"
+
+ AP_FLAKE8_CLEAN
+"""
+
+import os
+import subprocess
+import sys
+
+import argparse
+
+os.environ['PYTHONUNBUFFERED'] = '1'
+
+
+class AStyleChecker(object):
+    def __init__(self):
+        self.retcode = 0
+        self.directories_to_check = [
+            'libraries/AP_DDS',
+        ]
+        self.files_to_check = []
+
+    def progress(self, string):
+        print("****** %s" % (string,))
+
+    def check(self):
+        '''run astyle on all files in self.files_to_check'''
+        for path in self.files_to_check:
+            self.progress("Checking (%s)" % path)
+        astyle_command = ["astyle", "--dry-run"]
+        astyle_command.extend(self.files_to_check)
+        ret = subprocess.run(
+            astyle_command,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+            text=True
+        )
+        if ret.returncode != 0:
+            self.progress("astyle check failed: (%s)" % (ret.stdout))
+            self.retcode = 1
+        if "Formatted" in ret.stdout:
+            self.progress("Files needing formatting found")
+            self.retcode = 1
+
+    def find_files_in_directory(self, d):
+        '''append filepaths of .c and .h files under d to self.files_to_check'''
+        for (dirpath, dirnames, filenames) in os.walk(d):
+            for filename in filenames:
+                if os.path.splitext(filename)[1] not in [".cpp", ".c", ".h"]:
```

Comment:
Can you define this list at the top of the file for clarity?

Reply:
Pretty sure that just obfuscates things.